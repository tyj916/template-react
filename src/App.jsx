import 'normalize.css'
import './App.css'
import { useState } from 'react'

function App() {
  const [heading, setHeading] = useState("Hello World!");

  function clickHanlder() {
    setHeading('Success!');
  }

  return (
    <>
      <div>
        <h1>{heading}</h1>
      </div>
      <button type='button' onClick={clickHanlder}>
        Test Button
      </button>
    </>
  )
}

export default App
