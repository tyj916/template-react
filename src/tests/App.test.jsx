import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from '../App';

describe('App component', () => {
  it('renders hello world', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it('renders success after button click', async () => {
    const user = userEvent.setup();

    render(<App />);
    const button = screen.getByRole('button', {name: 'Test Button'});

    await user.click(button);

    expect(screen.getByRole('heading').textContent).toMatch(/success/i);
  });
});