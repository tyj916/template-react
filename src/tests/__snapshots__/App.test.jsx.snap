// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders hello world 1`] = `
<div>
  <div>
    <h1>
      Hello World!
    </h1>
  </div>
  <button
    type="button"
  >
    Test Button
  </button>
</div>
`;
